### DISCUSSION TITLE
when wire:click   causes disappearance of mapped item of a Collection

### DISCUSSION THREAD

**Discussion by FrankFlow** [livewire/livewire#7479] [7. Bugs (archived)]
### Livewire version

v3.2.6

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

I have a table with data. When I click a button to set something in Livewire Component a column of mapped data disappear.
![livewire](https://github.com/livewire/livewire/assets/7053026/2a4da9c8-e777-4d2a-8259-2fc27d03a1a6)

`  
[...]
->map(function ($item) {
                $item->aggiornamento = $item->updated_at->format('d-m-Y');
                return $item;
            });`

I have no "updated" method and foo() methods contains nothing. Only wire:click triggers this behaviour.

`    public function foo()
    {
       //do nothing
    }`

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@FrankFlow the issue is Livewire doesn't know how to hydrate Eloquent Collection's with custom joins, selects, maps, etc.

I go into this in detail in my recent LaraconAU talk and how to use computed properties instead. See here for more details https://www.youtube.com/watch?v=esD6o8HYV5Q&t=407s

Hope this helps!

**Reply by FrankFlow** [reply]
Thanks Josh, your workaround is fantastic. I'm trying to adapt your solution to my project: all my queries are in models.
Anyway, this Livewire issue worries me.
Thank you very much for you answer.

**Reply by joshhanley** [reply]
No worries 🙂
